pkg/router/template/router.go

```diff
@@ -171,6 +176,17 @@ func newTemplateRouter(cfg templateRouterCfg) (*templateRouter, error) {
 	return router, nil
 }
 
+func matchString(pattern, s string) bool {
```

Comment:
I'd rather not take a regex here, this is incredibly slow to allocate a new regex every item.  Should be `matchString(s string, allowedValues []string)`, since none of the examples you describe below really need full regex functionality.